SofaGLFW/src/SofaGLFW/SofaGLFWBaseGUI.cpp

```diff
@@ -489,23 +491,66 @@ void SofaGLFWBaseGUI::error_callback(int error, const char* description)
     SOFA_UNUSED(error);
     msg_error("SofaGLFWBaseGUI") << "Error: " << description << ".";
 }
- 
+
+int SofaGLFWBaseGUI::handleArrowKeys(int key)
+{
+    // Handling arrow keys with custom codes
+    switch (key)
+    {
+            case GLFW_KEY_UP: return 19;   // Custom code for up
+            case GLFW_KEY_DOWN: return 21; // Custom code for down
+            case GLFW_KEY_LEFT: return 18; // Custom code for left
+            case GLFW_KEY_RIGHT: return 20; // Custom code for right
+    }
+    // Default case return the given value as GLFW handle it
+    return key;
+}
+
 void SofaGLFWBaseGUI::key_callback(GLFWwindow* window, int key, int scancode, int action, int mods)
 {
-    SOFA_UNUSED(scancode);
+    const char keyName = SofaGLFWBaseGUI::handleArrowKeys(key);
+    int state = glfwGetKey(window, GLFW_KEY_LEFT_CONTROL);
 
+    auto currentGUI = s_mapGUIs.find(window);
+    if (currentGUI == s_mapGUIs.end() || currentGUI->second == nullptr)
+    {
+        return;
+    }
+
+    auto rootNode = currentGUI->second->getRootNode();
+    if (!rootNode)
+    {
+        return;
+    }
+    if (state == GLFW_PRESS)
+    {
+        if (action == GLFW_PRESS)
+        {
+            if (key==GLFW_KEY_LEFT_CONTROL)
```

Comment:
With the Qt GUI, you need to press CTRL AND the other key. Here, I suspect that your don't consider that both keys are pressed at the same time

Reply:
Well it has the same behavior as QT, The keyboard events are handled only if I press control+key

Reply:
ok thanks. I understand the code now